src/index.js

```diff
@@ -85,48 +85,58 @@ const GravityFormForm = ({ id, formData, lambda, presetValues = {} }) => {
-
-                    <FieldBuilder
-                        errors={errors}
-                        formData={singleForm}
-                        formId={id}
-                        presetValues={presetValues}
-                        register={register}
-                    />
-                    <button className="gravityform__button" type="submit">
-                        {singleForm.button.text
-                            ? singleForm.button.text
-                            : 'Submit'}{' '}
-                        {formLoading && (
-                            <span className="gravityform__button__loading_span">
-                                Loading
-                            </span>
+            <div className="gform_wrapper" id={`gform_wrapper_${id}`}>
+                <div className="gform_anchor" id={`gf_${id}`} />
+                {singleForm && (
+                    <form
+                        className={
+                            formLoading
+                                ? `gravityform gravityform--loading gravityform--id-${id}`
+                                : `gravityform gravityform--id-${id}`
+                        }
+                        //TODO: ID change go GF standard "gfrom_1"?
+                        id={`gravityform--id-${id}`}
+                        key={`gravityform--id-${id}`}
+                        onSubmit={handleSubmit(onSubmitCallback)}
+                    >
+                        {generalError && (
+                            <FormGeneralError errorCode={generalError} />
                         )}
-                    </button>
-                </form>
-            )
+                        <div className="gform_body">
+                            <ul
+                                className={`gform_fields form_sublabel_${singleForm.subLabelPlacement} description_${singleForm.descriptionPlacement} ${singleForm.labelPlacement}`}
+                                id={`gform_fields_${id}`}
+                            >
+                                <FieldBuilder
+                                    errors={errors}
+                                    formData={singleForm}
+                                    formId={id}
+                                    presetValues={presetValues}
+                                    register={register}
+                                />
+                            </ul>
+                        </div>
+
+                        <div
+                            className={`gform_footer ${singleForm.labelPlacement}`}
+                        >
+                            <input
+                                className="gravityform__button gform_button button"
+                                id={`gform_submit_button_${id}`}
+                                type="submit"
+                                value={`${singleForm?.button?.text ||
+                                    'Submit'}${formLoading ? ' Loading' : ''}`}
```

Comment:
Please can the loading message can be kept in a span. This is so it can be removed by styling/targeted to use CSS to add spinner loaders.

Reply:
Inputs can't have things inside them, but I'll see if this can be changed back to a button, if it can I'll revert this change.